### PR TITLE
Skip unknown file TLVs instead of dropping the transfer

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
## What

`BitchatFilePacket.decode` resolves every TLV tag through the four-value `TLVType` enum and bails on the first miss:

```kotlin
val t = TLVType.from(data[off].toUByte()) ?: return null
```

So a file packet carrying one tag this build does not know is not partially understood — it is discarded whole, **media included**. The receiver shows nothing and logs nothing; the sender sees a successful transfer. Both apps look perfectly healthy, and text in the same chat keeps working, which makes it a rough one to track down from either end.

iOS has always skipped unknown tags — `case nil: continue` in its own `BitchatFilePacket.decode`. So the two implementations disagree about what a valid packet is, and the tag list stopped being extensible in practice: any optional field added by a newer or third-party client costs every Android peer the entire file rather than just that field.

This PR makes Android match iOS. Unknown tags read a 2-byte length and skip the value. Known-tag handling, the 4-byte `CONTENT` length, multi-`CONTENT` concatenation, and every existing rejection (short buffer, `FILE_SIZE` length != 4, missing name/content) are unchanged.

## How we hit it

Building a bitchat-compatible client, we added one optional TLV (a message id, tag `0x05`, used only to key a delivery receipt) on the premise that unknown tags are skipped — which is true of the iOS decoder we read. Every image, voice note and file we sent to an Android peer then vanished silently. Worth noting the iOS repo already documents this asymmetry in `PrivateMediaMessageIdentity`:

> Android clients reject unknown file tags, so eligible senders and receivers derive the receipt key from fields already on the wire.

That workaround exists because of this line. Fixing the decoder means future optional fields degrade to "that field is ignored" instead of "the file is lost", which is what the TLV shape is for.

## Tests

Two added to `FileTransferTest`, both failing on the current decoder:

- `decode should skip unknown TLV types instead of dropping the whole file` — hand-built packet with an unknown `0x05` TLV between `MIME_TYPE` and `CONTENT` (where an encoder appending content last would put it); asserts name, mime, size and every content byte survive.
- `decode should skip an unknown TLV that trails the content` — same extension placed *after* `CONTENT`, which a decoder that stops at the first unknown tag also loses; asserts the `application/octet-stream` default and content still hold.

`./gradlew :app:testDebugUnitTest --tests "com.bitchat.FileTransferTest"` → 11 tests, 0 failures. Reverting the one-line decoder change fails exactly the two new tests.

Happy to adjust the naming/placement to match your conventions, or to add a matching test on the iOS side pinning the `case nil: continue` behaviour so the two decoders cannot drift apart again.

### CODE REVIEW

File: app/src/main/java/com/bitchat/android/model/BitchatFilePacket.kt [hunk 1]
```diff
@@ -122,6 +132,10 @@ data class BitchatFilePacket(
                                 contentBytes = (contentBytes!! + value)
                             }
                         }
+                        null -> android.util.Log.d(
+                            "BitchatFilePacket",
+                            "⏭️ Skipping unknown TLV type 0x${tag.toString(16)} ($len bytes)"
+                        )
```
[the remark this answers is not in the document]
Comment (vincenzopalazzo) [reply app/src/main/java/com/bitchat/android/model/BitchatFilePacket.kt]: **Fixed in 46de1a3.** Good catch -- the iteration count on that path is chosen by the sender, so per-TLV work is attacker-scaled, and a zero-length unknown TLV costs 3 bytes.

Two changes rather than a rate limit:

- Unknown tags now `continue` before `data.copyOfRange(off, off + len)`, so the skip path allocates nothing per TLV either -- the padded shape produced millions of empty-array copies on top of the log lines.
- The diagnostic is aggregated: one `Log.d` after the loop with the skipped count, emitted only when non-zero.

Added `decode should handle a packet padded with many zero-length unknown TLVs` (200k of them). Decodes in 15ms, vs 88ms with the per-TLV version -- and that ratio understates it, since Robolectric shadows `Log.d` into a near no-op while a real device pays a JNI write per line plus the logcat flooding you describe.

Worth noting the pre-existing `val value = data.copyOfRange(...)` on the known-tag path has the same sender-chosen shape (a zero-length `0x01` repeated), but that predates this PR and is bounded by the four known tags, so I left it alone. Happy to fold it in if you want it hardened here.
